### ISSUE TITLE
Daemon connect issues - Random

### ISSUE THREAD

**Issue by BKdilse** [turtlecoin/turtlecoin#305]
My Daemon is having problems connecting to Remote Daemons, and the problem is intermitent.  Other internet activity is unaffected, and I've run out of ideas.

Sometimes after a few restarts of the Daemon, it start working, other times, it takes several hours.

I can telnet into the remote daemon at the time of the issue, but the Daemon itself does not sync.  I am reporting this remotely, so difficult to capture debug logs.

My ISP has run tests, which have passed, I've reset everything, router, firewall, double checked port etc.

Any suggestions on pin pointing what is going on?

**Comment by RocksteadyTC** [comment]
Can you tell us more? Are you running turtlecoind-ha or any proxies?

**Comment by BKdilse** [comment]
Forgot to update this.  I have the problem right now.

Just running normal turtlecoind, no proxies.

Bits from the daemon console:

```
status
Height: 556933/557089 (99.97%) on mainnet, syncing, net hash 3.25 MH/s, v4, up to date, 8(out)+0(in) connections, uptime 0d 0h 53m 38s

print_cn
2018-Jun-21 21:57:06.811102 INFO    Connections:
2018-Jun-21 21:57:06.811102 INFO    Remote Host              Peer ID             Recv/Sent (inactive,sec) State                    Lifetime(seconds)
[OUTGOING]77.175.185.198:118974b0caa84b1c47f44    state_normal             1328
[OUTGOING]173.249.58.144:11897e8f87d9341369058    state_normal             1590
[OUTGOING]124.226.18.161:11897bfd0a1df6a68b1fc    state_synchronizing      1432
[OUTGOING]45.55.33.219:11897281e547afa4e3606    state_synchronizing      2327
[OUTGOING]92.21.166.157:118975dad2f83f4109443    state_synchronizing      311
[OUTGOING]192.169.82.178:118977837bf0c9fefee8     state_synchronizing      2544
[OUTGOING]76.8.166.41:1189743ff34543c2136cd    state_synchronizing      2643
[OUTGOING]109.65.130.217:118978d36dcf800656f1b    state_synchronizing      2004

Stopped and started a few times, and now it's bloody synching. :)

```

I don't think it's an issue with the code, but I've run out of ideas, so wanted to run it by you, as you guys are clued up on what the process is.

**Comment by BKdilse** [comment]
Will be offline now, but please let me know what may help in diagnosing this.

**Comment by BKdilse** [comment]
Will try and spend time on the weekend, to troubleshoot further.
When the problem occurs, I have tried using a VPN service on my Laptop, to test this, and it works each time I try.  I then disconnect the VPN, and on the same Laptop try the Daemon again, which fails to connect, but I guess it's connecting to random seeds, so may not prove much ???

Could my IP be banned on remote peers?
Could I just be very unlucky, and hitting outdated/buggy peers, hence why it eventualy starts synching?

**Comment by BKdilse** [comment]
Syncing fine for 2 days now on latest Daemon.

**Comment by BKdilse** [comment]
had this issue again today, and managed to get it working again, by deleting the DB, downloaded a compiled v0.6.3, and using checkpoints.csv.  It got all the way to height 500,133 and then just sat there.

Can someone please advise on what is going wrong here, and whether I can run any tests to see why it's hanging?

**Comment by BKdilse** [comment]
@RocksteadyTC hey Rock, tried that this morning, and just would not sync, so downloaded the pre-compiled version.  I'll see how this is in the morning.

**Comment by crappyrules** [comment]
@BKdilse I was able to get everything up and going by making some changes in an unmerged PR by @ZedPea and using the bootstrap from https://turtle.land/#chain.  If you want to give them a spin, I have the files already compiled either in this zip http://turtle.imhard4.men/trtl-0.6.4.tar.gz or you can download what you need here.  http://turtle.imhard4.men/trtl-0.6.4/

**Comment by BKdilse** [comment]
@crappyrules thanks.  I'm just trying to understand what is going wrong here. It worked fine until height 500k and then just stop, and not It a not synching at all.  Do you have details of with It stops working/connecting to peers?

Just want to make sure it's nothing on my end.

**Comment by ZedPea** [comment]
Can you increase logging to debug? I'm thinking you might see this in your logs: 

```
2018-Jul-08 20:45:20.969155 DEBUG   [173.255.214.106:11897 OUT] Exception in connectionHandler: SwappedVector::push_back
```

**Comment by crappyrules** [comment]
@BKdilse I've had that happen before but I was pressed for time so I just shut it down and pushed another backup into my server without digging into it.

**Comment by BKdilse** [comment]
@ZedPea I looked at it this morning, it wasn't syncing, so I killed it and started it again, and it started syncing.  When you say debug logging, is that the same as say `set_log 4` ?  (so I can grab details next time).

@crappyrules thanks for you suggestions, as it's now syncing, i'll leave it.  I really want to get to the bottom of why this is happening.  I have a test box, so if there is anything you guys want me to test out, I'm happy to do so.
